File: !old_files_drone_ground/timer/timer.cpp
#include "timer.h"

timer::timer() {
	LARGE_INTEGER li;
	if (!QueryPerformanceFrequency(&li)) {
		_errdbg.errorMSGwithGLE((LPTSTR)"QueryPerformanceFrequency");
	}
	PCFreq = double(li.QuadPart) / 1000.0;
}

void timer::StartCounter() {
	LARGE_INTEGER li;
	if (!QueryPerformanceCounter(&li)) {
		_errdbg.errorMSGwithGLE((LPTSTR)"QueryPerformanceCounter");
	}
	CounterStart = li.QuadPart;
}

double timer::GetCounter() {
	LARGE_INTEGER li;
	QueryPerformanceCounter(&li);
	return double(li.QuadPart - CounterStart) / PCFreq;
}

File: src/levtempfli/drone_ground/Timer.java
package levtempfli.drone_ground;

public class Timer {
    private long CounterStart = 0L;

    public void StartCounter() {
        CounterStart = System.currentTimeMillis();
    }

    public long GetCounter() {
        return System.currentTimeMillis() - CounterStart;
    }
}
